lance/lance-common/src/main/java/org/apache/gravitino/lance/common/ops/gravitino/GravitinoLanceNamespaceWrapper.java

```diff
@@ -136,39 +142,297 @@ public ListNamespacesResponse listNamespaces(
+            schemaName,
+            CommonUtil.formatCurrentStackTrace());
+      }
+    }
   }
 
   @Override
   public ListTablesResponse listTables(
       String id, String delimiter, String pageToken, Integer limit) {
     throw new UnsupportedOperationException("Not implemented yet");
   }
+
+  private boolean isLakehouseCatalog(Catalog catalog) {
+    return catalog.type().equals(Catalog.Type.RELATIONAL)
+        && "generic-lakehouse".equals(catalog.provider());
+  }
+
+  private Catalog loadAndValidateLakehouseCatalog(String catalogName) {
+    Catalog catalog;
+    try {
+      catalog = client.loadCatalog(catalogName);
+    } catch (NoSuchCatalogException e) {
+      throw LanceNamespaceException.notFound(
+          "Catalog not found: " + catalogName,
+          NoSuchCatalogException.class.getSimpleName(),
+          catalogName,
+          CommonUtil.formatCurrentStackTrace());
+    }
+    if (!isLakehouseCatalog(catalog)) {
+      throw LanceNamespaceException.notFound(
+          "Catalog not found: " + catalogName,
+          NoSuchCatalogException.class.getSimpleName(),
+          catalogName,
+          CommonUtil.formatCurrentStackTrace());
+    }
+    return catalog;
+  }
+
+  private CreateNamespaceResponse createOrUpdateCatalog(
+      String catalogName, CreateNamespaceRequest.ModeEnum mode, Map<String, String> properties) {
+    CreateNamespaceResponse response = new CreateNamespaceResponse();
+    try {
+      Catalog catalog = client.loadCatalog(catalogName);
+      // Catalog exists, handle based on mode
+      switch (mode) {
+        case EXIST_OK:
+          response.setProperties(Maps.newHashMap());
+          return response;
+        case CREATE:
+          throw LanceNamespaceException.conflict(
+              "Catalog already exists: " + catalogName,
+              CatalogAlreadyExistsException.class.getSimpleName(),
+              catalogName,
+              CommonUtil.formatCurrentStackTrace());
+        case OVERWRITE:
+          CatalogChange[] changes =
+              buildChanges(
+                  properties,
+                  catalog.properties(),
+                  CatalogChange::setProperty,
+                  CatalogChange::removeProperty,
+                  CatalogChange[]::new);
+          Catalog alteredCatalog = client.alterCatalog(catalogName, changes);
+          Optional.ofNullable(alteredCatalog.properties()).ifPresent(response::setProperties);
+          return response;
+        default:
+          throw new IllegalArgumentException("Unknown mode: " + mode);
+      }
+    } catch (NoSuchCatalogException e) {
+      // Catalog does not exist, create it
+      Catalog createdCatalog =
+          client.createCatalog(
+              catalogName, Catalog.Type.RELATIONAL, "generic-lakehouse", properties);
+      response.setProperties(
+          createdCatalog.properties() == null ? Maps.newHashMap() : createdCatalog.properties());
+      return response;
```

Comment:
I would suggest split the create and update logic. for example:

```
// load catalog
// if not existed, create the catalog
// if existed, update the catalog.
```

Rather than:

```
// load catalog
// update catalog.
// if failed, created the catalog.
```

Reply:
ok